src/main/kotlin/com/demonwav/mcdev/i18n/inspections/TranslationInspection.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.i18n.inspections
+
+import com.intellij.codeInspection.BaseJavaLocalInspectionTool
+import com.intellij.codeInspection.InspectionManager
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.JavaElementVisitor
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiElementVisitor
+import com.intellij.psi.PsiField
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiMethod
+
+abstract class TranslationInspection : BaseJavaLocalInspectionTool() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return object : JavaElementVisitor() {
+            override fun visitMethod(method: PsiMethod) {
+            }
+
+            override fun visitClass(aClass: PsiClass) {
+            }
+
+            override fun visitField(field: PsiField) {
+            }
+
+            override fun visitElement(element: PsiElement?) {
+                addDescriptors(checkElement(element, holder.manager, isOnTheFly))
```

Comment:
It's up to you and I know IntelliJ does it similar but I personally prefer to pass the `ProblemsHolder` directly into the inspections instead of first creating an array of `ProblemDescriptors` (you can check the mixin inspections for an example).  `ProblemsHolder` has many convenient shortcut methods to register problems without having to set `isOnTheFly` and others all the time)